python-sdk/src/astro/databases/snowflake.py

```diff
@@ -61,6 +61,80 @@
 NATIVE_LOAD_SUPPORTED_FILE_TYPES = (FileType.CSV, FileType.NDJSON, FileType.PARQUET)
 NATIVE_LOAD_SUPPORTED_FILE_LOCATIONS = (FileLocation.GS, FileLocation.S3)
 
+NATIVE_AUTODETECT_SCHEMA_SUPPORTED_FILE_TYPES = {FileType.PARQUET}
+NATIVE_AUTODETECT_SCHEMA_SUPPORTED_FILE_LOCATIONS = {FileLocation.GS, FileLocation.S3}
+
+
+@dataclass
+class SnowflakeFileFormat:
+    """
+    Dataclass which abstracts properties of a Snowflake File Format.
+
+    Snowflake File Formats are used to define the format of files stored in a stage.
+
+    Example:
+
+    .. code-block:: python
+
+        snowflake_stage = SnowflakeFileFormat(
+            name="file_format",
+            file_type="PARQUET",
+        )
+
+    .. seealso::
+        `Snowflake official documentation on file format creation
+        <https://docs.snowflake.com/en/sql-reference/sql/create-file-format.html>`_
+    """
+
+    name: str = ""
+    _name: str = field(init=False, repr=False, default="")
+    file_type: str = ""
+
+    @staticmethod
+    def _create_unique_name() -> str:
+        """
+        Generate a valid Snowflake file format name.
+
+        :return: unique file format name
+        """
+        return (
+            "file_format_"
```

Comment:
Looks like we are having a duplication of sorts, would it be a good idea to have a single function to generate a random string of the desired length and take prefix into account? 

https://github.com/astronomer/astro-sdk/blob/f1030fdeceed2d7de857d044c1e3773b05bf9ef3/python-sdk/src/astro/sql/table.py#L82

Reply:
Or we can use this function in table.py

Reply:
I don't have a strong preference here. But if we want to make it more elegant and truly unique, I would prefer using [uuid](https://docs.python.org/3/library/uuid.html) (maybe uuid4) to create a unique name and store the function in a utils file? WDYT?

Reply:
We can use it. But I guess we considered UUI for table names, but I cannot recall why we didn't. @tatiana might have better context there.